File: spaceinvaders/src/main/java/com/example/spaceinvaders/model/DTO/NaveDTO.java
package com.example.spaceinvaders.model.DTO;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class NaveDTO {
    
    String nombre;
    Long idEstrella;
    String nombreEstrella;
    Long idPlaneta;
    String nombrePlaneta;
    String tipoNave;
    Float creditos;
    Float tiempoRestante;
    Float capacidadMaxima;
    Float capadidadUtilizada;
    String foto;
    Float velocidad;

    public NaveDTO(String nombre, Long idEstrella, String nombreEstrella, Long idPlaneta, String nombrePlaneta, String tipoNave, Float creditos, Float tiempoRestante, Float capacidadMaxima, Float capadidadUtilizada, String foto, Float velocidad) {
        this.nombre = nombre;
        this.idEstrella = idEstrella;
        this.nombreEstrella = nombreEstrella;
        this.idPlaneta = idPlaneta;
        this.nombrePlaneta = nombrePlaneta;
        this.tipoNave = tipoNave;
        this.creditos = creditos;
        this.tiempoRestante = tiempoRestante;
        this.capacidadMaxima = capacidadMaxima;
        this.capadidadUtilizada = capadidadUtilizada;
        this.foto = foto;
        this.velocidad = velocidad;
    }

}

File: angular-client/src/app/model/nave-dto.ts
export class NaveDTO {
    constructor(
        public nombre: string,
        public idEstrella: number,
        public nombreEstrella: string,
        public idPlaneta: number,
        public nombrePlaneta: string,
        public tipoNave: string,
        public creditos: number,
        public tiempoRestante: number,
        public capacidadMaxima: number,
        public capadidadUtilizada: number,
        public foto: string,
        public velocidad: number
    ) {
    }
}
